### PR TITLE
Epics testing

### PR DESCRIPTION
**Pull request by jdetle**
Hello, all! It has been a while! I added some tests for the github-publish epic and refactored out some anonymous functions. 

Is there a better word for chore than writing tests? Sorry if its the wrong verb.

### CODE REVIEW

File: src/notebook/epics/github-publish.js [hunk 1]
```diff
@@ -163,31 +198,4 @@ export function handleGistAction(action, store) {
...
-        });
-        return;
-      }
-      notificationSystem.addNotification({
-        title: 'Unknown Publication Error',
-        message: err.toString(),
-        level: 'error',
-      });
-    });
+    .catch((err) => handleGistError(err, store));
```
Comment (rgbkrk) [inline src/notebook/epics/github-publish.js]: If you make the store the first argument you can create this anonymous function earlier and bind it:

const boundHandleGistError = handleGistError.bind(null, store)

Then pass boundHandleGistError to the catch.

**Review by rgbkrk — APPROVED** [review]

File: src/notebook/epics/github-publish.js (hunk 1 above)
Comment (jdetle) [reply src/notebook/epics/github-publish.js]: would that be inside the epic?

File: src/notebook/epics/github-publish.js [hunk 2]
```diff
@@ -135,6 +135,41 @@ export function publishNotebookObservable(github, notebook, filepath,
...
 /**
+ * Handle gist errors for the publish epic.
+ * @param  {String} error - Error response to be parsed and handled.
+ * @param  {Immutable.Record} store - Redux store containing current state.
+ *
+ */
+export function handleGistError(error, store) {
+  const state = store.getState();
+  const notificationSystem = state.app.get('notificationSystem');
+  // TODO: Let this go into the general error flow
```
Comment (rgbkrk) [inline src/notebook/epics/github-publish.js:145]: This highlights something important which is that we could be propagating these errors up as actions.

File: src/notebook/epics/github-publish.js (hunk 1 above)
Comment (rgbkrk) [reply src/notebook/epics/github-publish.js]: Yeah, before the actions of type pary

**Review by rgbkrk — APPROVED** [review]

**Comment by jdetle** [comment]
Shucks well this isn't working:

```
export const publishEpic = (action$, store) =>
  const boundHandleGistError = handleGistError.bind(null, store);
  action$.ofType(PUBLISH_USER_GIST, PUBLISH_ANONYMOUS_GIST)
    .mergeMap((action) => handleGistAction(action, store))
    .catch(boundHandleGistError(err));
```

I dont know if you wanted something else?
I got an error at `const boundHandleGistError`

**Comment by rgbkrk** [comment]
Turn

``` js
export const publishEpic = (action$, store) =>
  const boundHandleGistError = handleGistError.bind(null, store);
  action$.ofType(PUBLISH_USER_GIST, PUBLISH_ANONYMOUS_GIST)
    .mergeMap((action) => handleGistAction(action, store))
    .catch(boundHandleGistError(err));
```

into

``` js
export const publishEpic = (action$, store) => {
  const boundHandleGistError = handleGistError.bind(null, store);
  return action$.ofType(PUBLISH_USER_GIST, PUBLISH_ANONYMOUS_GIST)
    .mergeMap((action) => handleGistAction(action, store))
    .catch(boundHandleGistError(err));
}
```

The reason is that the arrow function without braces is an expression.

``` js
() => 2
```

is equivalent to

``` js
() => {
  return 2
}
```

As soon as you need more than one statement you have to use the braces and write an explicit return (if you need it):

``` js
export const publishEpic = (action$, store) => {
  const boundHandleGistError = handleGistError.bind(null, store);
  return action$.ofType(PUBLISH_USER_GIST, PUBLISH_ANONYMOUS_GIST)
    .mergeMap((action) => handleGistAction(action, store))
    .catch(boundHandleGistError(err));
}
```

**Review by rgbkrk — APPROVED** [review]

**Pull request merged**